### PR TITLE
WASM: Enable Invariant.Tests

### PR DESCRIPTION
**Pull request by safern**
Contributes towards: https://github.com/dotnet/runtime/issues/38422

### CODE REVIEW

File: eng/testing/tests.mobile.targets [hunk 1]
```diff
@@ -133,19 +133,6 @@
       <WasmFilesToIncludeInFileSystem Include="@(ReferenceCopyLocalPaths)" Condition="'%(ReferenceCopyLocalPaths.BuildReference)' == 'true' and !$([System.String]::new('%(ReferenceCopyLocalPaths.Identity)').EndsWith('.resources.dll'))" />
       <WasmFilesToIncludeInFileSystem Include="@(WasmSatelliteAssemblies)" TargetPath="%(WasmSatelliteAssemblies.CultureName)\%(WasmSatelliteAssemblies.Filename)%(WasmSatelliteAssemblies.Extension)" />
       <ExtraAssemblies Include="$(PublishDir)$(AssemblyName).dll" />
-      <!-- we need to preserve these facades for BinaryFormatter tests -->
```
Comment (safern) [inline eng/testing/tests.mobile.targets:136]: Apparently these facades are still needed.

File: src/libraries/Native/Unix/System.Globalization.Native/pal_icushim_static.c [hunk 2]
```diff
@@ -76,6 +79,12 @@ void GlobalizationNative_InitICUFunctions(void* icuuc, void* icuin, const char*
 
 int32_t GlobalizationNative_GetICUVersion(void)
 {
+    // this method is only used from our tests
+    // this way we ensure we're testing on the right mode
+    // even though we can call u_getVersion without loading since it is staticly linked.
```
Comment (danmosemsft) [inline src/libraries/Native/Unix/System.Globalization.Native/pal_icushim_static.c:84]: Typo statically

File: eng/testing/tests.mobile.targets (hunk 1 above)
Comment (akoeplinger) [reply eng/testing/tests.mobile.targets:136]: I looked at the test failures and it seems only keeping `mscorlib.dll` and `System.Drawing.dll` should be enough, and the latter is even only used by `System.Resources.Extensions.Tests.PreserializedResourceWriterTests.TypeConverterStringResources`.

Comment (safern) [reply eng/testing/tests.mobile.targets:136]: Yeah I'll keep those two. Thanks

**Comment by safern** [comment]
Alright, Browser leg is now green.

**Comment by SamMonoRT** [comment]
@safern @naricc trying to understand why we don't see the Interpreter lanes for runtime tests, we do see the other Mono Prio0 runtime test lanes.

**Comment by safern** [comment]
It is because in order to optimize resources and avoid dedicating a machine to invoke helix to run the interpreter tests and one for non-interpreter tests, we use the same lane for both and just send to helix jobs with different environment variable settings.

```
 Starting Azure Pipelines Test Run Mono Linux arm64 Release @ (Ubuntu.1804.Arm64.Open)Ubuntu.1804.Armarch.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm64v8-a45aeeb-20190620155855
  Starting Azure Pipelines Test Run Mono Linux arm64 Release interpreter @ (Ubuntu.1804.Arm64.Open)Ubuntu.1804.Armarch.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm64v8-a45aeeb-20190620155855
```

You can find that under:
Mono Pri0 Runtime Tests Run Linux arm64 release

In `Send to helix` step.

**Comment by naricc** [comment]
> @safern @naricc trying to understand why we don't see the Interpreter lanes for runtime tests, we do see the other Mono Prio0 runtime test lanes for jit though

We moved the interpreter to be a scenario instead of a seperate lane. So the interpreter is running, along with the jit, as part of this one "Mono Common Pri0 Test Build AnyOS AnyCPU release". I just double checked to verify it is still running the interpreter.

**Review by akoeplinger — APPROVED** [review]

**Comment by safern** [comment]
Badges didn't update.

**Pull request merged**